Source/DafnyStandardLibraries/src/Std_Concurrent/Std_Concurrent.go

```diff
@@ -114,23 +108,25 @@ func (_this *MutableMap) Get(k interface{}) Std_Wrappers.Option {
 }
 func (_this *MutableMap) Put(k interface{}, v interface{}) {
   {
-    _this.Internal.Store(k, v)
+    _this.mu.Lock()
+    _this.Internal = _this.Internal.UpdateUnsafe(k, v)
+    _this.mu.Unlock()
   }
 }
 func (_this *MutableMap) Remove(k interface{}) {
   {
-    _this.Internal.Delete(k)
+    // This could be special-cased for a single remove to be a bit faster,
+    // but it's still going to be O(n) so likely not worth it.
+    _this.mu.Lock()
+    _this.Internal = _this.Internal.Subtract(_dafny.SetOf(k))
```

Comment:
I assume reassignment is because `dafny.Map` is immutable? Why create a set out of `k` though?

Reply:
Just because `Subtract` already exists on `_dafny.Map`, so this was the easiest approach. I was trying to avoid depending on any runtime changes so it's easier to pick up this fix. As per the comment avoiding the singleton set is not worth it by itself. :)